mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp

```diff
@@ -57,6 +57,18 @@ static TosaOp createOpAndInfer(mlir::PatternRewriter &rewriter,
   return op;
 }
 
+static tosa::CastOp createCastOp(PatternRewriter &rewriter, Location loc,
+                                 Type resElementType, Value input) {
+  ShapedType inputType = input.getType().cast<ShapedType>();
```

Comment:
Minor question: does this need the `cast`? `getInput()` may well be a `TypedValue<ShapedType>` already.

Reply:
Unfortunately `input` is a Value, `getType()` will only return the most generic `mlir::Type`. There has to be a type cast there to grab the element type.